src/components/Login.tsx

```diff
@@ -1,19 +1,81 @@
 import type { Theme } from "@emotion/react";
 import { useTheme } from "@emotion/react";
 import { css } from "@emotion/react";
+import { useRef, useState } from "react";
 import { useNavigate } from "react-router";
 
 const Login = () => {
   const theme = useTheme();
   const navigate = useNavigate();
+  const ref = useRef(null);
+
+  const passwordValidator = {
+    test: (pw) => {
+      return pw.length >= 8;
+    },
+  };
+
+  const exp = /^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}$/;
```

Comment:
이 부분은 어떤 스트링을 식별하기 위한 정규표현식인지 드러나도록 하면 좋을 거 같습니다.

예시
```suggestion
  const EMAIL_REGEXP = /^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}$/;
```